src/sentry/monitors/endpoints/base.py

```diff
@@ -151,3 +142,59 @@ def convert_args(
 
         kwargs.update({"checkin": checkin})
         return args, kwargs
+
+
+class MonitorIngestEndpoint(MonitorEndpoint):
+    """
+    This type of endpont explicitly only allows for DSN and Token / Key based authentication.
+
+    [!!]: These endpoints are legacy and will be replaced by relay based
+          checkin ingestion in the very near future.
+    """
+
+    # TODO(epurkhiser): This should not extend MonitorEndpoint, that base class
+    # is designed for frontend API endpoints
+
+    authentication_classes = (DSNAuthentication, TokenAuthentication, ApiKeyAuthentication)
+
+    # TODO(dcramer): this code needs shared with other endpoints as its security focused
+    # TODO(dcramer): this doesnt handle is_global roles
+    def convert_args(
+        self,
+        request: Request,
+        monitor_id: str,
+        checkin_id: str | None = None,
+        *args,
+        **kwargs,
+    ):
+        args, kwargs = super().convert_args(request, monitor_id, *args, **kwargs)
+
+        monitor = kwargs["monitor"]
+
+        if checkin_id is None:
+            return args, kwargs
```

Comment:
This is so that we can use this base endpoint class for the `monitor_ingest_checkin_index` endpoint module, which doesn't need to lookup a checkin.